test/trento/domain/host/host_test.exs

```diff
@@ -644,4 +648,66 @@ defmodule Trento.HostTest do
       )
     end
   end
+
+  describe "deregistration" do
+    test "should emit the HostDeregistered Event" do
+      host_id = Faker.UUID.v4()
+      dat = DateTime.utc_now()
+
+      host_registered_event = build(:host_registered_event, host_id: host_id)
+
+      assert_events(
+        [host_registered_event],
+        [
+          %DeregisterHost{
+            host_id: host_id,
+            deregistered_at: dat
+          }
+        ],
+        %HostDeregistered{
+          host_id: host_id,
+          deregistered_at: dat
+        }
+      )
+    end
+
+    test "should emit the HostDeregistrationRequest Event" do
+      host_id = Faker.UUID.v4()
+
+      host_registered_event = build(:host_registered_event, host_id: host_id)
+
+      assert_events(
+        [host_registered_event],
+        [
+          %RequestHostDeregistration{
+            host_id: host_id
+          }
+        ],
+        %HostDeregistrationRequested{
+          host_id: host_id
+        }
+      )
+    end
+
+    test "should apply the HostDeregistered command and set the deregistration date into the state" do
+      host_id = Faker.UUID.v4()
+      dat = DateTime.utc_now()
+
+      host_registered_event = build(:host_registered_event, host_id: host_id)
+
+      assert_state(
+        [
+          host_registered_event,
+          %HostDeregistered{
+            host_id: host_id,
+            deregistered_at: dat
+          }
+        ],
+        [],
+        fn host ->
+          assert host.deregistered_at == dat
```

Comment:
Please use `expected == actual` in assertions. This way if it fails we know that the left part of the diff is the expected one.